docs/src/main/sphinx/language/types.md

```diff
@@ -41,31 +41,58 @@ This type captures boolean values `true` and `false`.
 
 ## Integer
 
+Integer numbers can be expressed as numeric literals in the following formats:
+
+* Decimal integer. Examples are `-7`, `0`, or `3`.
+* Hexadecimal integer composed of `0X` or `0x` and the value. Examples are
+  `0x0A` for decimal `10` or `0x11` for decimal `17`.
+* Octal integer composed of `0O` or `0o` and the value. Examples are `0o40` for
+  decimal `32` or `0o11` for decimal `9`.
+* Binary integer composed of `0B` or `0b` and the value. Examples are `0b1001`
+  for decimal `9` or `0b101010` for decimal `42``.
+
+Underscore characters are ignored within literal values, and can be used to
+increase readability. For example, decimal integer `123_456.789_123` is
+equivalent to `123456.789123`. Preceding underscores are not permitted.
+Preceding underscores and trailing underscores are not permitted.
+
+Integers are supported by the following data types with increasing size range.
+
 ### `TINYINT`
 
 A 8-bit signed two's complement integer with a minimum value of
-`-2^7` and a maximum value of `2^7 - 1`.
+`-2^7` or `-0x80` and a maximum value of `2^7 - 1` or `0x7F`.
 
 ### `SMALLINT`
 
 A 16-bit signed two's complement integer with a minimum value of
-`-2^15` and a maximum value of `2^15 - 1`.
+`-2^15` or `-0x8000` and a maximum value of `2^15 - 1` or `0x7FFF`.
 
-### `INTEGER`
+### `INTEGER` or `INT`
 
-A 32-bit signed two's complement integer with a minimum value of
-`-2^31` and a maximum value of `2^31 - 1`.  The name `INT` is
-also available for this type.
+A 32-bit signed two's complement integer with a minimum value of `-2^31` or
+`-0x80000000` and a maximum value of `2^31 - 1` or `0x7FFFFFFF`.  The names
+`INTEGER` and `INT` can both be used for this type.
 
 ### `BIGINT`
 
-A 64-bit signed two's complement integer with a minimum value of
-`-2^63` and a maximum value of `2^63 - 1`.
+A 64-bit signed two's complement integer with a minimum value of `-2^63` or
+`-0x8000000000000000` and a maximum value of `2^63 - 1` or `0x7FFFFFFFFFFFFFFF`.
 
 (floating-point-data-types)=
 
 ## Floating-point
 
+Variable-precision numbers can be expressed as numeric literal using the
+scientific notation such as `1.03e1` for the `DOUBLE` data type. The supported
+`REAL` and `DOUBLE` data types also support explicit declaration of the type
```

Comment:
> also support explicit declaration of the type with the keywords `REAL` or `DOUBLE` and a value.

What does this mean? If this is referring to the `DOUBLE '1.23e4'` syntax. There's nothing special about these types -- that's just the generic type constructor syntax, and it applies to every type unless the type has no textual representation.

Reply:
well... if you dont have the type defined .. you get double automatically .. how do we say that  .. or do we just leave it as implied from the following sections?

Reply:
Those are two different things. Numeric scientific precision *literals* have a type of `DOUBLE`. E.g.,

```
1.23e4
```

Values of `DOUBLE` and `REAL` (and most other types) can be constructed via their type constructor:

```
DOUBLE '123e4'
REAL '123e4'
VARCHAR 'abcde'
DECIMAL '123.4'
TIMESTAMP '2023-08-21 12:34:56'
etc.
```

Reply:
I think I updated this now so it makes more sense and is clear.